core/src/main/java/org/elasticsearch/index/mapper/core/StringFieldMapper.java

```diff
@@ -146,6 +146,27 @@ public StringFieldMapper build(BuilderContext context) {
         @Override
         public Mapper.Builder parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
             StringFieldMapper.Builder builder = stringField(name);
+            // hack for the fact that string can't just accept true/false for
+            // the index property and still accepts no/not_analyzed/analyzed
+            final Object indexO = node.remove("index");
+            if (indexO != null) {
+                final String index = Strings.toUnderscoreCase(indexO.toString());
+                switch (index) {
+                case "analyzed":
+                    builder.tokenized(true);
+                    node.put("index", true);
+                    break;
+                case "not_analyzed":
+                    builder.tokenized(false);
+                    node.put("index", true);
+                    break;
+                case "no":
+                    node.put("index", false);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Can't parse [index] value [" + index + "], expected [true], [false], [no], [not_analyzed] or [analyzed]");
```

Comment:
Should the exception message contain the input `index0` instead of the normalized `index`?